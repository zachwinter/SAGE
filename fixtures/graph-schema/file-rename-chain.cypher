// Test case: File renamed A.ts → B.ts → C.ts across commits
// Expected: WAS_RENAMED_FROM relationships linking the chain

MATCH (c:File {name: "C.ts"})-[:WAS_RENAMED_FROM]->(b:File {name: "B.ts"})-[:WAS_RENAMED_FROM]->(a:File {name: "A.ts"})
WHERE c.first_seen = 3 AND b.first_seen = 2 AND a.first_seen = 1
RETURN c.name, b.name, a.name;