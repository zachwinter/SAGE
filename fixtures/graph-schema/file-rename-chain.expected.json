[
  {
    "c.name": "C.ts",
    "b.name": "B.ts",
    "a.name": "A.ts"
  }
]
